src/structures/GuildScheduledEvent.js

```diff
@@ -0,0 +1,325 @@
+'use strict';
+
+const Base = require('./Base');
+const {
+  GuildScheduledEventEntityTypes,
+  GuildScheduledEventStatuses,
+  GuildScheduledEventPrivacyLevels,
+} = require('../util/Constants');
+const SnowflakeUtil = require('../util/SnowflakeUtil');
+
+/**
+ * Represents a scheduled event in a {@link Guild}.
+ * @extends {Base}
+ */
+class GuildScheduledEvent extends Base {
+  constructor(client, data) {
+    super(client);
+
+    /**
+     * The id of the guild scheduled event
+     * @type {Snowflake}
+     */
+    this.id = data.id;
+
+    /**
+     * The id of the guild this guild scheduled event belongs to
+     * @type {Snowflake}
+     */
+    this.guildId = data.guild_id;
+
+    /**
+     * Whether the guild scheduled event has been deleted
+     * @type {boolean}
+     */
+    this.deleted = false;
+
+    this._patch(data);
+  }
+
+  _patch(data) {
+    /**
+     * The channel id in which the scheduled event will be hosted, or `null` if entity type is `EXTERNAL`
+     * @type {?Snowflake}
+     */
+    this.channelId = data.channel_id;
+
+    /**
+     * The id of the user that created this guild scheduled event
+     * @type {?Snowflake}
+     */
+    this.creatorId = data.creator_id;
+
+    /**
+     * The name of the guild scheduled event
+     * @type {string}
+     */
+    this.name = data.name;
+
+    if ('description' in data) {
+      /**
+       * The description of the guild scheduled event
+       * @type {?string}
+       */
+      this.description = data.description;
+    } else {
+      this.description ??= null;
+    }
+
+    /**
+     * The timestamp the guild scheduled event will start at
+     * @type {number}
+     */
+    this.scheduledStartTime = data.scheduled_start_time;
+
+    /**
+     * The timestamp the guild scheduled event will end at,
+     * or `null` if the event does not have a scheduled time to end
+     * @type {?number}
+     */
+    this.scheduledEndTime = data.scheduled_end_time;
```

Comment:
These are ISO strings, not numbers. Can you convert these to timestamps in ms?

Reply:
good catch